sched/semaphore/sem_post.c

```diff
@@ -259,16 +259,18 @@ int nxsem_post(FAR sem_t *sem)
    * else try to get it in slow mode.
    */
 
-#if !defined(CONFIG_PRIORITY_INHERITANCE) && !defined(CONFIG_PRIORITY_PROTECT)
-  if (sem->flags & SEM_TYPE_MUTEX)
+  if ((sem->flags & SEM_TYPE_MUTEX)
```

Comment:
@jlaitine please include some comments here after changing the code, the explanation from commit message could be use as hint here.